src/core/network.js

```diff
@@ -414,9 +415,27 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition =
+      fullRequestXhr.getResponseHeader('Content-Disposition');
+    this._fileName = this._extractNameFrom(contentDisposition);
+
     this._headersReceivedCapability.resolve();
   },
 
+  _extractNameFrom:
+      function PDFNetworkStreamFullRequestReader_extractNameFrom(cDisposition) {
+    if (cDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(cDisposition);
```

Comment:
I'm not a contributor, just patiently waiting for this pull.

I think the point @jsperezg is making is that if validating the file name falls within the scope of his work, then it should also raise an exception so the user is not confused. I could imagine a developer beating their head against a wall for a good many hours because a validation event on user input is failing silently.

I don't really think validating the file name is within scope, or really necessary at all, but like I said I'm not a contributor - my opinion doesn't matter in that regard.

Thanks for the good work @jsperezg!